### PR TITLE
Fix compile errors on rawhide (g++8.0.1)

### PR DESCRIPTION
**Pull request by AlanGriffiths**

### CODE REVIEW

**Comment by AlanGriffiths** [comment]
@Conan-Kudo I'm not entirely sure how to build my working copy with the options used for the archive builds, so I may have missed something. If you could check?

(NB gmock still seems broken)

File: src/client/display_configuration.cpp [hunk 1]
```diff
@@ -55,7 +55,11 @@ mcl::DisplayOutput::DisplayOutput(size_t num_modes_, size_t num_formats)
 
 mcl::DisplayOutput::DisplayOutput(DisplayOutput&& rhs)
 {
-    std::memcpy(this, &rhs, sizeof(*this));
+    std::memcpy(
+        static_cast<MirDisplayOutput*>(this),
+        static_cast<MirDisplayOutput*>(&rhs),
+        sizeof(MirDisplayOutput));
```
Comment (RAOF) [inline src/client/display_configuration.cpp]: Hm. This is invoking undefined behaviour, isn't it? I can see why GCC 8 is complaining about this - we're std::memcpy()ing something that isn't TriviallyCopyable (among other things, it has a non-trivial move constructor *right here* 😁 ), but casting *this into the base class which is TriviallyCopyable doesn't resolve the problem, it just silences the warning, while still letting GCC 8 optimise this code into `rm -rf /`?

This is *somewhat* academic - while there's no guarantee that mcl::DisplayOutput has the same memory-representation as MirDisplayOutput it quite obviously will. Unless we later change mcl::DisplayOutput and forget to update the move constructor.

File: src/server/glib_main_loop_sources.cpp [hunk 2]
```diff
@@ -431,9 +431,9 @@ void md::FdSources::add(
 
     g_source_set_callback(
         gsource,
-        reinterpret_cast<GSourceFunc>(&FdContext::static_call),
+        (GSourceFunc)(void*)(&FdContext::static_call),
         fd_context,
-        reinterpret_cast<GDestroyNotify>(&FdContext::static_destroy));
+        (GDestroyNotify)(void*)(&FdContext::static_destroy));
```
Comment (RAOF) [inline src/server/glib_main_loop_sources.cpp]: Ok, I think I see why this complains.

Shouldn't this be reinterpret_cast<GDestroyNotify>(reinterpret_cast<void*>(&FdContext::static_destroy))?

File: src/client/display_configuration.cpp (hunk 1 above)
Comment (AlanGriffiths) [reply src/client/display_configuration.cpp]: This is fine, we're copying the TriviallyCopyable base member. If there were an implicit_cast<> we would use it. static_cast<> is the next best thing.

File: src/server/glib_main_loop_sources.cpp (hunk 2 above)
Comment (AlanGriffiths) [reply src/server/glib_main_loop_sources.cpp]: OK, verbosity is in. ;)

**Comment by RAOF** [comment]
Oooooh, right. It doesn't matter how DisplayOutput is laid out!

Although it's still a rake that can be stepped on if mcl::DisplayOutput ever becomes something more than a trivial wrapper.

But since we're not *planning* to touch mirclient again, that should be fine.

**Pull request merged**